tests/components/ruckus_unleashed/test_device_tracker.py

```diff
@@ -0,0 +1,103 @@
+"""The sensor tests for the Ruckus Unleashed platform."""
+from datetime import timedelta
+
+from homeassistant.components.ruckus_unleashed import DOMAIN
+from homeassistant.const import CONF_MAC, STATE_HOME, STATE_NOT_HOME, STATE_UNAVAILABLE
+from homeassistant.helpers import entity_registry
+from homeassistant.util import utcnow
+
+from tests.async_mock import patch
+from tests.common import async_fire_time_changed
+from tests.components.ruckus_unleashed import (
+    DEFAULT_TITLE,
+    TEST_CLIENT,
+    TEST_CLIENT_ENTITY_ID,
+    init_integration,
+    mock_config_entry,
+)
+
+
+async def test_client_connected(hass):
+    """Test client connected."""
+    await init_integration(hass)
+
+    future = utcnow() + timedelta(minutes=60)
+    with patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.clients",
+        return_value={
+            TEST_CLIENT[CONF_MAC]: TEST_CLIENT,
+        },
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        await hass.helpers.entity_component.async_update_entity(TEST_CLIENT_ENTITY_ID)
+        test_client = hass.states.get(TEST_CLIENT_ENTITY_ID)
+        assert test_client.state == STATE_HOME
+
+
+async def test_client_disconnected(hass):
+    """Test client disconnected."""
+    await init_integration(hass)
+
+    future = utcnow() + timedelta(minutes=60)
+    with patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.clients",
+        return_value={},
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        await hass.helpers.entity_component.async_update_entity(TEST_CLIENT_ENTITY_ID)
+        test_client = hass.states.get(TEST_CLIENT_ENTITY_ID)
+        assert test_client.state == STATE_NOT_HOME
+
+
+async def test_clients_update_failed(hass):
+    """Test failed update."""
+    await init_integration(hass)
+
+    future = utcnow() + timedelta(minutes=60)
+    with patch(
+        "homeassistant.components.ruckus_unleashed.Ruckus.clients",
+        side_effect=ConnectionError,
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+        await hass.helpers.entity_component.async_update_entity(TEST_CLIENT_ENTITY_ID)
+        test_client = hass.states.get(TEST_CLIENT_ENTITY_ID)
+        assert test_client.state == STATE_UNAVAILABLE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6309587)